### PR TITLE
[RAPTOR-5607 | HOTFIX] Bandaid issue when using -cd and --docker

### PR DESCRIPTION
**Pull request by jmbannon**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
When using `-cd` instead of `--code-dir` with `--docker`, `drum fit` can fail within docker.
Using `-cd` works locally. I've spent a few hours trying to figure out what is wrong with no avail. This hotfix will at least make `-cd` work from an outsider perspective, but the underlying issue still remains.

When using -cd, custom.py is not found within docker:
Screenshot from 2021-06-17 11-57-32

## Changes
Replace `-cd` with `--code-dir` when constructing the docker command

### CODE REVIEW

**Comment by eric-s-s** [comment]
i _think_ this is bad directory shenanigans:

custom_model_runner/datarobot_drum/drum/drum.pyL798

```
        CMRunnerUtils.replace_cmd_argument_value(in_docker_cmd_list, "-cd", in_docker_model)
```

and what that does is:

replace `"-cd"` in the command list with: `"/opt/model"`  _but it doesn't do this for --code-dir_

**Comment by eric-s-s** [comment]
nope! i was wrong.  the exact same thing is right above it for code-dir:
```
        CMRunnerUtils.replace_cmd_argument_value(
            in_docker_cmd_list, ArgumentsOptions.CODE_DIR, in_docker_model
        )
```

**Comment by eric-s-s** [comment]
i suspect that passing two arguments like this is an incorrect usage:
```
            parser.add_argument(
                "-cd",
                ArgumentsOptions.CODE_DIR,
                default=None,
                required=True,
                type=type_callback,
                help=help_message,
            )
```

and that replacing the command in docker is failing silently here:
datarobot_drum.drum.utils.CMRunnerUtils.replace_cmd_argument_value
```
    def replace_cmd_argument_value(cls, cmd_list, arg_name, arg_value):
        try:
            ind = cmd_list.index(arg_name)
        except ValueError:
            return
        cmd_list[ind + 1] = arg_value
```
but who knows.

**Review by eric-s-s — APPROVED** [review]
:shrug: if it works it works.

**Comment by yakov-g** [comment]
maybe let's just remove support for `-cd` ?

**Comment by jmbannon** [comment]
> maybe let's just remove support for `-cd` ?

I'd be okay with that, no other args have short versions

**Comment by yakov-g** [comment]
> > maybe let's just remove support for `-cd` ?
> 
> I'd be okay with that, no other args have short versions

Cool,
Are you going to do this?

**Comment by jmbannon** [comment]
@yakov-g New PR is here: https://github.com/datarobot/datarobot-user-models/pull/368

**Comment by jmbannon** [comment]
jarvis test functional_training_model_templates

File: custom_model_runner/CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Show full R traceback for fit/predict errors
 ##### Fixed
 - Loading hooks from both custom.R and custom.r
+- Using -cd instead of --code-dir failed sometimes when running with docker
```
Comment (4dahalibut) [inline custom_model_runner/CHANGELOG.md:12]: maybe better to be more official here. sometimes -> occasionally

Comment (jmbannon) [reply custom_model_runner/CHANGELOG.md:12]: Will fix in the release PR. Should remove 'sometimes' because it happens every time

**Review by scottp-ml — APPROVED** [review]

**Pull request merged**